### PR TITLE
feat: global ethclient

### PR DESCRIPTION
**Pull request by HenryQW**

### CODE REVIEW

File: service/indexer/internal/datasource/alchemy/datasource.go [hunk 1]
```diff
@@ -230,20 +228,11 @@ func (d *Datasource) getAssetTransactionHashes(ctx context.Context, message *pro
...
+		rpcClientMap: map[string]*alchemy.Client{},
 	}
 
 	var err error
 
-	if internalDatasource.ethereumClientMap[protocol.NetworkEthereum], err = ethclient.Dial(config.General.Ethereum.HTTP); err != nil {
-		return nil, err
-	}
-
-	if internalDatasource.ethereumClientMap[protocol.NetworkPolygon], err = ethclient.Dial(config.General.Polygon.HTTP); err != nil {
```
Comment (kallydev) [inline service/indexer/internal/datasource/alchemy/datasource.go:243]: Polygon also uses alchemy.

File: service/hub/internal/server/server.go [hunk 2]
```diff
@@ -88,6 +89,15 @@ func (s *Server) Initialize() (err error) {
 
 	cache.ReplaceGlobal(redisClient)
 
+	ethereumClientMap, err := ethclientx.Dial(config.ConfigHub.RPC, []string{protocol.NetworkEthereum, protocol.NetworkPolygon})
```
Comment (kallydev) [inline service/hub/internal/server/server.go]: The `ResolveCrossbell` function uses the Crossbell RPC, but the Hub does not initialize it.

```diff
func ResolveCrossbell(input string) (string, error) {
 	var result string
- 	ethereumClient, err := ethclient.Dial(config.ConfigHub.RPC.General.Crossbell.HTTP)
+ 	ethereumClient, err := ethclientx.Global(protocol.NetworkCrossbell)
 	if err != nil {
 		return "", fmt.Errorf("failed to connect to crossbell rpc: %s", err)
 	}
}
```

File: service/indexer/internal/worker/exchange/swap/worker_test.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ var (
 )
 
 func TestBuildTransferMetadata(t *testing.T) {
-	ethereumClientMap, err := ethereum.New(&configx.RPC{
+	ethereumClientMap, err := ethclientx.Dial(&configx.RPC{
```
Comment (kallydev) [inline service/indexer/internal/worker/exchange/swap/worker_test.go:36]: Missing networks parameter?

File: service/indexer/internal/worker/social/crossbell/worker.go [hunk 4]
```diff
@@ -46,7 +48,7 @@ func (s *service) Networks() []string {
 }
 
 func (s *service) Initialize(ctx context.Context) (err error) {
-	if s.ethereumClient, err = ethclient.Dial(Endpoint); err != nil {
+	if s.ethereumClient, err = ethclientx.Global(s.Networks()[0]); err != nil {
```
Comment (kallydev) [inline service/indexer/internal/worker/social/crossbell/worker.go]: ~Why not use the global ethereum client?~ Why not just use the global ethereum client?

File: service/indexer/internal/worker/exchange/swap/worker.go [hunk 5]
```diff
@@ -260,31 +259,9 @@ func (s *service) Jobs() []worker.Job {
 	}
 }
 
-func New(config *configx.RPC, employer *shedlock.Employer) (worker.Worker, error) {
-	var err error
-
+func New(employer *shedlock.Employer) (worker.Worker, error) {
 	svc := service{
-		ethereumClientMap: make(map[string]*ethclient.Client),
```
Comment (kallydev) [inline service/indexer/internal/worker/exchange/swap/worker.go:267]: `ethereumClientMap` still exists within the `service` structure.

**Review by kallydev — CHANGES_REQUESTED** [review]

File: service/indexer/internal/worker/social/crossbell/worker.go (hunk 4 above)
Comment (HenryQW) [reply service/indexer/internal/worker/social/crossbell/worker.go]: handler still needs to initialize with an ethclient, trying the keep the changes to the minimal 👀

File: service/indexer/internal/datasource/alchemy/datasource.go (hunk 1 above)
Comment (HenryQW) [reply service/indexer/internal/datasource/alchemy/datasource.go:243]: it was dialed after line 243, github didn't show them 👀

**Review by brucexc — APPROVED** [review]

**Pull request merged**